packages/@sanity/cli/src/commands/functions/envFunctionsCommand.ts

```diff
@@ -1,68 +1,60 @@
-import {type types} from '@sanity/runtime-cli/utils'
-
 import {type CliCommandDefinition} from '../../types'
 
-type StackFunctionResource = types.StackFunctionResource
-
 const helpText = `
-Arguments
-  [add] Add or update an environment variable
-  [remove] Remove an environment variable
+Commands
+  add    Add or update an environment variable
+  list   List the environment variables
+  remove Remove an environment variable
 
-Options
-  --name <name> The name of the function
-  --key <key> The name of the environment variable
-  --value <value> The value of the environment variable
+Arguments
+  <name> The name of the function
+  <key> The name of the environment variable
+  <value> The value of the environment variable
 
 Examples
   # Add or update an environment variable
-  sanity functions env add --name echo --key API_URL --value https://api.example.com/
+  sanity functions env add echo API_URL https://api.example.com/
 
   # Remove an environment variable
-  sanity functions env remove --name echo --key API_URL
-`
+  sanity functions env remove echo API_URL
 
-const defaultFlags = {
-  name: '',
-  key: '',
-  value: '',
-}
+  # List environment variables
+  sanity functions env list echo
+`
 
 const envFunctionsCommand: CliCommandDefinition = {
   name: 'env',
   group: 'functions',
   helpText,
-  signature: '',
-  description: 'Add or remove an environment variable for a Sanity function',
-  hideFromHelp: true,
+  signature: '<add|list|remove> <name> [key] [value]',
+  description:
+    'Add or remove an environment variable or list environment variables for a Sanity function',
   async action(args, context) {
     const {apiClient, output} = context
     const {print} = output
-    const [subCommand] = args.argsWithoutOptions
-    const flags = {...defaultFlags, ...args.extOptions}
+    const [subCommand, name, key, value] = args.argsWithoutOptions
 
-    if (!subCommand || !['add', 'remove'].includes(subCommand)) {
-      throw new Error('You must specify if you wish to add or remove an environment variable')
+    if (!subCommand || !['add', 'list', 'remove'].includes(subCommand)) {
+      throw new Error('You must specify if you want to list, add or remove')
     }
 
```

Comment:
We should probably do some checking to make sure all the arguments are provided here. Something like:

```javascript
if (subCommand === "add" && args.argsWithoutOptions.length < 4) {
  throw new Error("You must specify the name, key and value")
} else if (subCommand === "remove" && args.argsWithoutOptions.length < 3) {
  throw new Error("You must specify the name and key")
}
```

Don't need to check `list` as it only requires a `name` and that is handled on line 46.